tongueroo/jets
# fix guesser look for long function names

This is a 🐞 bug fix.




## Summary

Long controller and or function names result in a bug in the prewarming function call lookup logic.

## Context

* #445 
* https://community.rubyonjets.com/t/cloudformation-error-either-stackname-or-physicalresourceid-must-be-specified/424/4

## How to Test

Here's how to reproduce:

Example class:

```ruby
class UpController < ApplicationController
  def long_function_name_to_try_to_reproduce_an_prewarm_error
    render json: {success: true}
  end
end
```

Adjust the prewarming rate to see bug quickly:

```ruby
Jets.application.configure do
  config.prewarm.rate = '30 minutes' # default is '30 minutes'
  # ...
end
```

Deploy the code with the fix, the log should not crash at that function.

    2020-03-27 00:39:47 UTC 2020/03/27/[$LATEST]5527b8448ae342b0b876dcec3a4377ce Calling lambda function demo-dev-UpController-OAO-LongFunctionNameToTryToR-S48CLWB3YTQY on AWS

Believe this was introduced https://github.com/tongueroo/jets/pull/395 

## Version Changes

Minor
